test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -979,10 +979,10 @@ public void afterAll(ExtensionContext context) throws Exception {
     }
 
     private void runAfterAllCallbacks(ExtensionContext context) throws Exception {
-        if (isNativeOrIntegrationTest(context.getRequiredTestClass()) || failedBoot) {
-            return;
-        }
-        if (afterAllCallbacks.isEmpty()) {
+        if (failedBoot
```

Comment:
I don't really understand this, throwBootFailureException() in your stack trace will only throw if failedBoot is true, so how does this help? I also can't really see a code path where `runningQuarkusApplication==null` and failedBoot is false.

Is there any chance you could investigate this more to figure out how it gets into this state?

Reply:
I'll try. Btw, the method above `runAfterAllCallbacks()` (that is callling it) also has such a null check.

It helped in that I was able to see the actual failure and not just a misleading NPE.

Reply:
I think that check is for the integration test case.

Reply:
Aha, that `failedBoot` is not there in 2.2, which is the proper fix, as you mentioned! I'll recheck current `main`.

Reply:
Alright, before creating this PR I was distracted by https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/main.3A.20CNFE.20ConfigClassBuildItem and stopped right there.
If I ignore that problem, boot dev mode and run the tests, I can see the `ClassNotFoundException` right away, no NPE!

So this problem only exists in 2.2, so I will create a dedicated PR for that.